### PR TITLE
Migrate code from invoke to invoke_batch.

### LINKED ISSUE #13238: Migrate code to use ScalarUDF::invoke_batch and ScalarUDFImpl::invoke_batch
Migrate code from ScalarUDF::invoke and invoke_no_arg to use ScalarUDF::invoke_batch 
And from ScalarUDFImpl::invoke and invoke_no_arg  ScalarUDFImpl::invoke_batch.

The places in the  code to be changed can be found by searching for `#[allow(deprecated)] // TODO migrate UDF invoke to invoke_batch`

- follows https://github.com/apache/datafusion/pull/13001 -- introduced `invoke_batch`
- follows https://github.com/apache/datafusion/pull/13179 -- deprecated the other invoke variants for ScalarUDF
- follows https://github.com/apache/datafusion/pull/13174 -- deprecated the other invoke variants for ScalarUDFImpl

### PR DESCRIPTION
**Pull request by irenjj**
## Which issue does this PR close?

Closes #13238

## What changes are included in this PR?

Migrate code in unit test from invoke to invoke_batch.

## Are these changes tested?

yes

## Are there any user-facing changes?

no

### CODE REVIEW

**Comment by irenjj** [comment]
PTAL @findepi

File: datafusion/functions/src/datetime/date_bin.rs [hunk 1]
```diff
@@ -491,150 +491,197 @@ mod tests {
...
+                ColumnarValue::Scalar(ScalarValue::IntervalDayTime(Some(
+                    IntervalDayTime {
+                        days: 0,
+                        milliseconds: 1,
+                    },
+                ))),
+                ColumnarValue::Scalar(ScalarValue::TimestampNanosecond(Some(1), None)),
+                ColumnarValue::Scalar(ScalarValue::TimestampNanosecond(Some(1), None)),
+            ],
+            0,
```
Comment (findepi) [inline datafusion/functions/src/datetime/date_bin.rs:506]: would 1 be more suitable?

(same in other places)

**Review by jayzhan211 — APPROVED** [review]

**Comment by alamb** [comment]
Thank you very much @irenjj and @findepi and @jayzhan211 -- I merged this branch up from main to run CI against the latest and greatest versions

**Comment by findepi** [comment]
Thank you @irenjj @jayzhan211  @alamb .
it looks ready to merge

**Pull request merged**